djstripe/models/base.py

```diff
@@ -157,22 +159,25 @@ def _get_stripe_account_id(self, api_key=None) -> Optional[str]:
         reverse_account_relations = (
             field
             for field in self._meta.get_fields(include_parents=True)
-            if field.is_relation and field.one_to_many
-            # Avoid circular import problems by using the app registry to
-            # get the model class rather than a direct import.
-            and field.related_model
-            is apps.get_model(app_label="djstripe", model_name="account")
+            if field.is_relation
+            and field.one_to_many
+            and field.related_model is Account
         )
 
         # Handle case where we have a reverse relation to Account and should pass
         # that account ID to the retrieve call.
         for field in reverse_account_relations:
             # Grab the related object, using the first one we find.
             reverse_lookup_attr = field.get_accessor_name()
-            account = getattr(self, reverse_lookup_attr).first()
-
-            if account is not None:
-                return account.id
+            try:
+                account = getattr(self, reverse_lookup_attr).first()
+            except ValueError:
+                if isinstance(self, Account):
```

Comment:
else?

Reply:
I don't understand the question. Try loops have an else clause as well that only execute if the stuff in the try loop executes without any error.

Reply:
You're catching valueerror. For anything that is not an account, you're basically doing the equivalent of a `pass`.

So, what happens in the else case, what is supposed to?

Reply:
Yes, that's correct. 

I audited the code and there were 2 ways to fix this issue:

1) Re write the algorithm for django 4+

2) A hot fix that works in the majority of cases


I chose to do the latter. In case this method is called on a model other than Account and that model is not created in the db yet, the account_id will eventually get populated (set as None after first sync) as this would be the case of platform account. This is because on the next sync the account_id will get populated correctly and will be set as none after the first sync. Also this will not affect any of the stripe retrieve or API calls as stripe_account=None is automatically assumed by stripe to be on the platform account. 

I wanted to come up with a better method myself but there doesn't seem to be a simple solution because for the sync chain of a model to get saved in the DB, we first save all the FKs on the initial model (which is where this is called) but "this" requires the initial model to be saved which creates a chicken-egg situation. I suppose one thing I could try is to try to populate the "djstripe_owner_account" field in the _attach_objects_post_save_hook. What do you think? 

Also this is not relevant for connected accounts as we pass the stripe_account key explicitly so this search happens **only** in platform accounts.